tests/raft_log_replication.py

```diff
@@ -593,7 +593,441 @@ async def _unittest_raft_log_replication() -> None:
+    assert raft_node_1._log[3].entry.value == 9
+    assert raft_node_1._commit_index == 3
+
+    # check if the new entry is replicated in the follower nodes
+    assert len(raft_node_2._log) == 1 + 3
+    assert raft_node_2._log[0].term == 0
+    assert raft_node_2._log[0].entry.name.value.tobytes().decode("utf-8") == ""  # index zero entry is empty
+    assert raft_node_2._log[0].entry.value == 0
+    assert raft_node_2._log[1].term == 4
+    assert raft_node_2._log[1].entry.name.value.tobytes().decode("utf-8") == "top_1"
+    assert raft_node_2._log[1].entry.value == 7
+    assert raft_node_2._log[2].term == 5
+    assert raft_node_2._log[2].entry.name.value.tobytes().decode("utf-8") == "top_2"
+    assert raft_node_2._log[2].entry.value == 8
+    assert raft_node_2._log[3].term == 6
+    assert raft_node_2._log[3].entry.name.value.tobytes().decode("utf-8") == "top_3"
+    assert raft_node_2._log[3].entry.value == 9
+    assert raft_node_2._commit_index == 3
+
+    assert len(raft_node_3._log) == 1 + 3
+    assert raft_node_3._log[0].term == 0
+    assert raft_node_3._log[0].entry.name.value.tobytes().decode("utf-8") == ""  # index zero entry is empty
+    assert raft_node_3._log[0].entry.value == 0
+    assert raft_node_3._log[1].term == 4
+    assert raft_node_3._log[1].entry.name.value.tobytes().decode("utf-8") == "top_1"
+    assert raft_node_3._log[1].entry.value == 7
+    assert raft_node_3._log[2].term == 5
+    assert raft_node_3._log[2].entry.name.value.tobytes().decode("utf-8") == "top_2"
+    assert raft_node_3._log[2].entry.value == 8
+    assert raft_node_3._log[3].term == 6
+    assert raft_node_3._log[3].entry.name.value.tobytes().decode("utf-8") == "top_3"
+    assert raft_node_3._log[3].entry.value == 9
+    assert raft_node_3._commit_index == 3
+
+    assert len(raft_node_4._log) == 1 + 3
+    assert raft_node_4._log[0].term == 0
+    assert raft_node_4._log[0].entry.name.value.tobytes().decode("utf-8") == ""  # index zero entry is empty
+    assert raft_node_4._log[0].entry.value == 0
+    assert raft_node_4._log[1].term == 4
+    assert raft_node_4._log[1].entry.name.value.tobytes().decode("utf-8") == "top_1"
+    assert raft_node_4._log[1].entry.value == 7
+    assert raft_node_4._log[2].term == 5
+    assert raft_node_4._log[2].entry.name.value.tobytes().decode("utf-8") == "top_2"
+    assert raft_node_4._log[2].entry.value == 8
+    assert raft_node_4._log[3].term == 6
+    assert raft_node_4._log[3].entry.name.value.tobytes().decode("utf-8") == "top_3"
+    assert raft_node_4._log[3].entry.value == 9
+    assert raft_node_4._commit_index == 3
+    
+    await asyncio.sleep(TERM_TIMEOUT + 1)
+    
+    _logger.info("================== TEST 2: Leadership Change to Node 42 and Add New Entry  ==================")
+
+    # New LEADER => raft_node_2
+    raft_node_2.election_timeout = ELECTION_TIMEOUT
+    raft_node_3.election_timeout = ELECTION_TIMEOUT + 2
+    raft_node_4.election_timeout = ELECTION_TIMEOUT + 2
```

Comment:
I corrected the test [conditions](https://github.com/OpenCyphal-Garage/cyraft/pull/11/commits/2093d422adfd05814e9cc0daa470ca8c26429d86#:~:text=raft_node_1.election_timeout,2.1). But node 4 increased by +2.1, because node 2 launched the _serve_request_vote_impl method and node 4 launched the _start_election method in parallel, which is why the `self._voted_for` value was assigned the value of node 4 instead of None, which is why the request did not pass [the condition](https://github.com/OpenCyphal-Garage/cyraft/pull/11/commits/024fd95896d70b8fa74c7c8a52d69c40c4a24b76#:~:text=vote_granted%20%3D%20(self._voted_for%20is%20None%20or%20self._voted_for%20%3D%3D%20client_node_id)%20and%20self._log%5Brequest.last_log_index%5D.term%20%3D%3D%20request.last_log_term) and node 2 lost the election.